modules/Core/pages/panel/users_reports.php

```diff
@@ -89,20 +89,25 @@
             $comments = $queries->getWhere('reports_comments', ['report_id', '=', $report->id]);
             $comments = count($comments);
 
-            $target_user = new User($report->reported_id);
-
-            if ($report->type == Report::ORIGIN_WEBSITE) {
-                // Site report
-                $user_reported = $target_user->getDisplayname();
-                $user_profile = URL::build('/panel/user/' . urlencode($report->reported_id . '-' . $target_user->data()->username));
-                $user_style = $target_user->getGroupClass();
-                $user_avatar = $target_user->getAvatar();
-            } else {
-                // API report
+            $user_reported = null;
+            if ($report->reported_id != 0) {
+                $reported_user = new User($report->reported_id);
+
+                if ($reported_user->exists()) {
+                    // Reported user exists
+                    $user_reported = $reported_user->getDisplayname();
+                    $user_profile = URL::build('/panel/user/' . urlencode($report->reported_id . '-' . $reported_user->data()->username));
+                    $user_style = $reported_user->getGroupClass();
+                    $user_avatar = $reported_user->getAvatar();
+                }
+            }
+
+            if ($user_reported == null) {
+                // Reported user doesn't exist, use their username and uuid
                 $user_reported = Output::getClean($report->reported_mcname);
                 $user_profile = URL::build('/panel/user/' . urlencode($report->reported_id . '-' . $report->reported_mcname));
                 $user_style = '';
-                $user_avatar = $report->reported_id == 0 ? null : AvatarSource::getAvatarFromUUID($report->reported_uuid);
+                $user_avatar = AvatarSource::getAvatarFromUUID($report->reported_uuid);
```

Comment:
we cannot guarantee they will have a uuid, would be better to use `getAvatarFromUserData`, i think?

Reply:
Discussed in discord, already only used if user don't exist, but i did use username atleast if uuid is missing